core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -202,6 +202,7 @@ class ReplicaManager(val config: KafkaConfig,
     valueFactory = Some(tp => HostedPartition.Online(Partition(tp, time, this)))
   )
   private val replicaStateChangeLock = new Object
+  val replicaFetcherManager = createReplicaFetcherManager(metrics, time, threadNamePrefix, quotaManagers.follower)
```

Comment:
Do we need to create both a ReplicaFetcherManager and AsyncReplicaFetcherManager if we are only going to use one of them?

This change might be a little cleaner if you extracted an interface from ReplicaFetcherManager that just contained the functions used by ReplicaManager. Then you would only have one condition in this patch, which is where the fetcher manager is created.

Reply:
Thanks for the review! That's a good point, and I've updated the PR with the extracted interface.